### DISCUSSION TITLE
Update old environment

### DISCUSSION THREAD

**Discussion by EmileVinsonneau** [ravendb/ravendb#15790] [Q&A]
Hello everyone,

I am working on .Net Framework 4.7.2 project with RavenDB3.5.9 on visual studio 2017. And, we decide to update this project to .netcore6 and RavenDB5.4.100.

Well, we decide to migrate database client firstly then Microsoft framework. So, I create a new project with last version of RavenDB, then I create unit test to migrate code (1) and I make a configuration to load assembly dynamically (2) in order to swap "Raven3Provider" and "Raven5Provider".

(1) But when I run my unitTest, I have the following error that I don't understand :

`Exception thrown: 'System.ArgumentException' in System.Core.dll
Additional information: Le champ d'instance '_flags' n'est pas défini pour le type 'System.Decimal'`

with call stack :

`>	System.Core.dll!System.Linq.Expressions.Expression.Field(System.Linq.Expressions.Expression expression, string fieldName) + 0xde bytes	
 	Sparrow.dll!Sparrow.Utils.DecimalHelper.CreateIsDoubleMethod() + 0x73 bytes	
 	Sparrow.dll!Sparrow.Utils.DecimalHelper.DecimalHelper() + 0x36 bytes	
 	Sparrow.dll!Sparrow.Utils.DecimalHelper.DecimalHelper() + 0x2d bytes	
 	[Native to Managed Transition]	
 	[Managed to Native Transition]	
 	Sparrow.dll!Sparrow.Json.JsonOperationContext.WriteInternal(Sparrow.Json.AbstractBlittableJsonTextWriter writer, object json) + 0x4b bytes	
 	Sparrow.dll!Sparrow.Json.JsonOperationContext.Write(Sparrow.Json.AbstractBlittableJsonTextWriter writer, Sparrow.Json.Parsing.DynamicJsonValue json) + 0x31 bytes	
 	Raven.Client.dll!Raven.Client.Http.ClusterTopologyLocalCache.TrySavingAsync() + 0x321 bytes	
`

(2) when I load dynamically, I load dependencies then I load sparrow and Raven (5.4.100), but I have loader exceptions but exceptions show dependencies that I loaded just before. I don't understand why because I have the same version. 

Well, I have few experience with assembly loader. I readed it is not the best thing to do, but how I can swap between theses two versions.

Thanks

**Comment by ayende** [comment]
You are using RavenDB 5.4 client to with .NET 4.7.2 ? And getting this error?
This is surprising, since the relevant code is from 2016, and working properly.



Can you show the error in a stand alone project?

For loading dynamically, you'll need to post the full execption messages.

**Reply by EmileVinsonneau** [reply]
Thank to answer.

Finally, my priority change. I'll create a ticket to create a stand alone project...

For the point 2, I have a System.IO.FileNotFoundException, but I realize that system try to link with path from AppBase, that is not the good one.

**Comment by EmileVinsonneau** [comment]
We upgrade our IDE and we have no more error during UnitTests.

About my second error, I solved my error to add subscribe of AppDomain.CurrentDomain.AssemblyResolve that search in the right folder.
